lib/finch/pool_manager.ex

```diff
@@ -26,26 +23,43 @@ defmodule Finch.PoolManager do
     end
   end
 
-  def lookup_pool(key) do
-    case Registry.lookup(PoolRegistry, key) do
+  def lookup_pool(registry, key) do
+    case Registry.lookup(registry, key) do
       [] ->
         :none
 
       [{pid, _}] ->
         {:ok, pid}
+
+      pids ->
+        {pid, _} = Enum.random(pids)
```

Comment:
This is good for now. In the future, I think it would be nice to be able to plug in alternative strategies here.  But it's not something we need to worry about now.